### PR TITLE
Fixing clippy iotedged

### PR DESCRIPTION
**Pull request by huguesBouvier**
Fixing Clippy issues

### CODE REVIEW

**Review by lfitchett — DISMISSED** [review]

File: edgelet/dps/src/registration.rs [hunk 1]
```diff
@@ -192,15 +192,13 @@ where
         operation_id: &str,
         token_source: Option<DpsTokenSource<K>>,
     ) -> Box<dyn Future<Item = Option<DeviceRegistrationResult>, Error = Error> + Send> {
-        let c = if let Some(ts) = token_source {
+        let c = token_source.map_or(client.read().expect("RwLock read failure").clone(), |ts| {
```
Comment (arsing) [inline edgelet/dps/src/registration.rs:195]: `.map_or_else`

File: edgelet/edgelet-iothub/src/lib.rs [hunk 2]
```diff
@@ -303,8 +303,14 @@ where
     fn update(&mut self, id: IdentitySpec) -> Self::UpdateFuture {
         let module_id = id.module_id().to_string();
 
-        let result = if let Some(generation_id) = id.generation_id() {
-            match self.get_key_pair(&module_id, generation_id) {
+        let result = id.generation_id().map_or(
```
Comment (arsing) [inline edgelet/edgelet-iothub/src/lib.rs:306]: `.map_or_else`

File: edgelet/iotedge/src/check/checks/connect_management_uri.rs [hunk 3]
```diff
@@ -47,11 +47,10 @@ impl ConnectManagementUri {
             .diagnostics_image_name
             .starts_with("/azureiotedge-diagnostics:")
         {
-            if let Some(upstream_hostname) = settings.parent_hostname() {
-                upstream_hostname.to_string() + &check.diagnostics_image_name
-            } else {
-                "mcr.microsoft.com".to_string() + &check.diagnostics_image_name
-            }
+            settings.parent_hostname().map_or(
```
Comment (arsing) [inline edgelet/iotedge/src/check/checks/connect_management_uri.rs:50]: `.map_or_else`

File: edgelet/iotedge/src/check/checks/container_connect_upstream.rs [hunk 4]
```diff
@@ -96,11 +96,10 @@ impl ContainerConnectUpstream {
             .diagnostics_image_name
             .starts_with("/azureiotedge-diagnostics:")
         {
-            if let Some(upstream_hostname) = settings.parent_hostname() {
-                upstream_hostname.to_string() + &check.diagnostics_image_name
-            } else {
-                "mcr.microsoft.com".to_string() + &check.diagnostics_image_name
-            }
+            settings.parent_hostname().map_or(
```
Comment (arsing) [inline edgelet/iotedge/src/check/checks/container_connect_upstream.rs:99]: `.map_or_else`

File: edgelet/iotedge/src/check/checks/container_local_time.rs [hunk 5]
```diff
@@ -45,11 +45,10 @@ impl ContainerLocalTime {
             .diagnostics_image_name
             .starts_with("/azureiotedge-diagnostics:")
         {
-            if let Some(upstream_hostname) = settings.parent_hostname() {
-                upstream_hostname.to_string() + &check.diagnostics_image_name
-            } else {
-                "mcr.microsoft.com".to_string() + &check.diagnostics_image_name
-            }
+            settings.parent_hostname().map_or(
```
Comment (arsing) [inline edgelet/iotedge/src/check/checks/container_local_time.rs:48]: `.map_or_else`

File: edgelet/iotedge/src/check/checks/container_resolve_parent_hostname.rs [hunk 6]
```diff
@@ -40,11 +40,10 @@ impl ContainerResolveParentHostname {
             .diagnostics_image_name
             .starts_with("/azureiotedge-diagnostics:")
         {
-            if let Some(upstream_hostname) = settings.parent_hostname() {
-                upstream_hostname.to_string() + &check.diagnostics_image_name
-            } else {
-                "mcr.microsoft.com".to_string() + &check.diagnostics_image_name
-            }
+            settings.parent_hostname().map_or(
```
Comment (arsing) [inline edgelet/iotedge/src/check/checks/container_resolve_parent_hostname.rs:43]: `.map_or_else`

**Review by arsing — CHANGES_REQUESTED** [review]

File: edgelet/dps/src/registration.rs (hunk 1 above)
Comment (huguesBouvier) [reply edgelet/dps/src/registration.rs:195]: Fixed.

File: edgelet/edgelet-iothub/src/lib.rs (hunk 2 above)
Comment (huguesBouvier) [reply edgelet/edgelet-iothub/src/lib.rs:306]: Fixed.

File: edgelet/iotedge/src/check/checks/connect_management_uri.rs (hunk 3 above)
Comment (huguesBouvier) [reply edgelet/iotedge/src/check/checks/connect_management_uri.rs:50]: Fixed.

File: edgelet/iotedge/src/check/checks/container_local_time.rs (hunk 5 above)
Comment (huguesBouvier) [reply edgelet/iotedge/src/check/checks/container_local_time.rs:48]: Fixed.

File: edgelet/iotedge/src/check/checks/container_connect_upstream.rs (hunk 4 above)
Comment (huguesBouvier) [reply edgelet/iotedge/src/check/checks/container_connect_upstream.rs:99]: Fixed.

File: edgelet/iotedge/src/check/checks/container_resolve_parent_hostname.rs (hunk 6 above)
Comment (huguesBouvier) [reply edgelet/iotedge/src/check/checks/container_resolve_parent_hostname.rs:43]: Fixed.

File: edgelet/edgelet-iothub/src/lib.rs (hunk 2 above)
Comment (arsing) [reply edgelet/edgelet-iothub/src/lib.rs:306]: Hasn't been fixed.

**Review by arsing — CHANGES_REQUESTED** [review]
.

File: edgelet/edgelet-iothub/src/lib.rs (hunk 2 above)
Comment (huguesBouvier) [reply edgelet/edgelet-iothub/src/lib.rs]: didn't get pushed. Fixed.

**Review by arsing — APPROVED** [review]

**Pull request merged**